src/main/java/org/fenixedu/academic/service/services/enrollment/shift/WriteStudentAttendingCourse.java

```diff
@@ -40,7 +40,16 @@ protected void run(Registration registration, String executionCourseId) throws F
         if (registration == null) {
             throw new FenixServiceException("error.registration.not.exist");
         }
-        registration.addAttendsTo(readExecutionCourse(executionCourseId));
+
+        ExecutionCourse executionCourse = readExecutionCourse(executionCourseId);
+
+        for (Registration registration2 : registration.getStudent().getAllRegistrations()) {
```

Comment:
The method getAllRegistrations() is very heavy from a performance point-of-view.
If you look at the implementation of this method you will find that it constructs an unmodifiable collection based on an unmodifiable set that was constructed based on a new hash set that was filled with a filtered group of registrations. That's allot of unnecessary overhead. I suggest using the getRegistrationStream() method instead.